ui/components/ui/unit-input/unit-input.component.js

```diff
@@ -86,6 +86,10 @@ export default class UnitInput extends PureComponent {
   };
 
   handleInputBlur = ({ target: { value } }) => {
+    setTimeout(() => {
+      this.unitInput.scrollLeft = this.unitInput.scrollWidth;
+    }, 0);
+
```

Comment:
what does this do? 

seeing some odd behavior when I enter a long number like `0.12414314125153525326236`. only the ending numbers are displayed on unfocus. also happens when I hit Max and my balance has a lot of trailing numbers

![Screenshot 2024-06-07 at 3 35 49 PM](https://github.com/MetaMask/metamask-extension/assets/100321200/bb24f9a5-e1fb-4701-8cf3-66e7978dd571)